R/splitByRanks.R

```diff
@@ -121,37 +116,53 @@ setGeneric("splitByRanks",
     SimpleList(ans)
 }
 
-#' @rdname splitByRanks
+#' @rdname agglomerate-methods
 #' @export
-setMethod("splitByRanks", signature = c(x = "SummarizedExperiment"),
-    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, ...){
+setMethod("agglomerateByRanks", signature = c(x = "SummarizedExperiment"),
+    function(x, ranks = taxonomyRanks(x), na.rm = TRUE, as.list = FALSE, ...){
+        browser()
         args <- .norm_args_for_split_by_ranks(na.rm = na.rm, ...)
-        .split_by_ranks(x, ranks, args)
+        if( !.is_a_bool(as.list) ){
+            stop("'as.list' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if( as.list ){
+            .split_by_ranks(x, ranks, args)
+        }
+        else{
+            altExps(x) <- .split_by_ranks(x, ranks, args)
+            x
+        }
```

Comment:
Store result to variable and end the function with return(x) --> tidier

Reply:
Also do this:

x <- .split_by_ranks(x, ranks, args)
if( !as.list ){
    # Add to altexps
}
return(x)

Reply:
That is simpler

Reply:
Moreover, you are overwriting existing altExps which is problem. Instead, append the altExps list

Reply:
This is method for SE. It does not have altExps so you have to convert it to TreeSE

Reply:
> Also do this:
> 
> x <- .split_by_ranks(x, ranks, args) if( !as.list ){ # Add to altexps } return(x)

If I do that, won't the method return a SimpleList no matter the value of `as.list`? The value returned by .split_by_ranks internal function is a SimpleList if I am not mistaken. Do we want to return a SimpleList even when the result is added to altExps

Reply:
Yes, you are correct. How about

```
res <- .split_by_ranks()
# Add to altExp if user has specified to do so
if( !as.list ){
    # Create a general function that adds results to altExps (That could be used in other functions also in the future)
    res <- .add_to_altExps(x, res)
}
return(res)
```

That might be easier to read